dspace-api/src/main/java/org/dspace/browse/BrowsableObject.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.browse;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.dspace.core.Constants;
+
+/**
+ * This is the basic interface that a data model entity need to implement to support browsing/retrieval
+ * 
+ * @author Andrea Bollini (andrea.bollini at 4science.it)
+ *
+ * @param <PK>
+ *            the Class of the primary key
+ */
+public interface BrowsableObject<PK extends Serializable> {
+    Map<String, Object> extraInfo = new HashMap<String, Object>();
+
+    /**
+     * A map of additional information exposed by the Entity to the browse/retrieve system
+     * 
+     * @return a map of extra information, by default an empty map is returned
+     */
+    default public Map<String, Object> getExtraInfo() {
+        return extraInfo;
+    }
+
+    /**
+     * 
+     * @return true if the Entity should be considered finalized (archived in terms of DSpace Item)
+     */
+    default public boolean isArchived() {
```

Comment:
It's also unclear to me why we should be replicating both `isArchived()` and `isDiscoverable()` from the `Item` class.  These seem to *only* apply the Items (and not any other object).  Couldn't we consider dropping these and simply have `SolrServiceImpl` check if the `BrowseableObject` is also an `Item` (and if so, cast it to an Item)? 

Again, if both of these are now required for all BrowseableObjects, I think it'd be helpful to explain why via JavaDocs.  Otherwise, this will be entirely unclear to the next developer who looks at this code.

Reply:
@tdonohue method removed as agreed via slack